### PR TITLE
Fixed delayed power state update after vm shutdown

### PR DESCRIPTION
**Pull request by Spaceman1984**
## Description

After a vm is shutdown, the power state isn't updated immediately. This prevents changing the service offering.
This PR updates the power state immediately after the vm is confirmed to be shutdown.

Fixes: #3159 
## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

This has been tested by starting a vm, waiting for the power state in the db to be updated to powerOn, then shutting down the vm and as soon as the vm is down, checking the power state in the db. Can also be verified by changing the service offering after stopping the vm.

### CODE REVIEW

**Comment by Spaceman1984** [comment]
@blueorangutan package

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 1]
```diff
@@ -1657,6 +1657,10 @@ protected boolean sendStop(final VirtualMachineGuru guru, final VirtualMachinePr
                     return false;
                 }
 
+                final UserVmVO userVm = _userVmDao.findById(vm.getId());
+                userVm.setPowerState(PowerState.PowerOff);
+                _userVmDao.update(userVm.getId(), userVm);
+
                 guru.finalizeStop(profile, answer);
```
Comment (rhtyd) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:1664]: @Spaceman1984 should we update the power state after guru finalises the stop? What is there is an exception then the power state will be off while VM may actually be running?

Comment (Spaceman1984) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:1664]: np @rhtyd, I'll move it down.

**Comment by Spaceman1984** [comment]
@blueorangutan package

**Comment by Spaceman1984** [comment]
@blueorangutan test ?

**Comment by Spaceman1984** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by rhtyd** [comment]
@blueorangutan test

**Review by rhtyd — APPROVED** [review]

**Review by wido — APPROVED** [review]
LGTM

Sounds logical!

**Pull request merged**